dds/DCPS/transport/shmem/ShmemSendStrategy.cpp

```diff
@@ -41,16 +41,19 @@ ShmemSendStrategy::start_i()
   const size_t n_elems = datalink_control_size_ / sizeof(ShmemData),
     extra = datalink_control_size_ % sizeof(ShmemData);
 
-  void* mem = alloc->calloc(datalink_control_size_);
-  if (mem == 0) {
+  void* mem = 0;
+  if (alloc == 0 || (mem = alloc->calloc(datalink_control_size_)) == 0) {
     VDBG_LVL((LM_ERROR, "(%P|%t) ERROR: ShmemSendStrategy for link %@ failed "
               "to allocate %B bytes for control\n", link_, datalink_control_size_), 0);
     return false;
   }
 
   ShmemData* data = reinterpret_cast<ShmemData*>(mem);
-  data[(extra >= sizeof(int)) ? n_elems : (n_elems - 1)].status_ =
-    SHMEM_DATA_END_OF_ALLOC;
+  const size_t limit = (extra >= sizeof(int)) ? n_elems : (n_elems - 1);
+  for (size_t i = 0; i < limit; ++i) {
+    data[i].status_ = SHMEM_DATA_FREE;
```

Comment:
Is this needed?  I thought the `calloc` above was taking care of it.

Reply:
Ah, yeah, I see now that calloc has a default initial_value parameter set to '0', which should match FREE.